kpi/utils/django_orm_helper.py

```diff
@@ -38,13 +38,19 @@ def __init__(self, expression: str, keyname: str, increment: int, **extra):
         )
 
 
-class OrderRandom(Func):
+class OrderCustom(Func):
 
     function = 'array_position'
     template = '%(function)s(ARRAY%(order_list)s, %(expressions)s)'
     arity = 1
 
-    def __init__(self, expression: str, order_list: list, **extra):
+    def __init__(self, expression: str, order_list: list, array_type: str, **extra):
```

Comment:
array_type: str is a required argument. It's allowed to be `Literal['varchar']` to get the desired varchar behavior else it will use the implicit cast. Thus to use implicit cast, it must be called with `array_type='unicorns'` to function as None isn't accepted.

How about making the type `Optional[literal['varchar']] = None`

If we're going to place the type hint, we should as well make the type useful. Then it can be omitted to use implicit cast and mypy would error if set to unicorns, which is silly.